cloudant-client/src/test/java/com/cloudant/tests/BulkDocumentTest.java

```diff
@@ -15,39 +15,20 @@
 package com.cloudant.tests;
 
 import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertThat;
+import static org.hamcrest.MatcherAssert.assertThat;
 
-import com.cloudant.client.api.Database;
 import com.cloudant.client.api.model.Response;
 import com.cloudant.test.main.RequiresDB;
-import com.cloudant.tests.util.CloudantClientResource;
-import com.cloudant.tests.util.DatabaseResource;
+import com.cloudant.tests.base.TestWithDb;
 import com.google.gson.JsonObject;
 
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.rules.RuleChain;
+import org.junit.jupiter.api.Test;
 
 import java.util.ArrayList;
 import java.util.List;
 
-@Category(RequiresDB.class)
-public class BulkDocumentTest {
-
-    public static CloudantClientResource clientResource = new CloudantClientResource();
-    public static DatabaseResource dbResource = new DatabaseResource(clientResource);
-    @ClassRule
-    public static RuleChain chain = RuleChain.outerRule(clientResource).around(dbResource);
-
-
-    private static Database db;
-
-    @BeforeClass
-    public static void setUp() {
-        db = dbResource.get();
-    }
+@RequiresDB
+public class BulkDocumentTest extends TestWithDb {
```

Comment:
Couldn't `TestWithDb` reasonably include the `@RequiresDB` annotation and avoid the need for it to be present throughout so many test classes. I'm assuming anything that extends `TestWithDb` also `RequiresDB` though I guess which DB could be important. Something to think about.

Reply:
It's a nice idea but we have test classes like `CloudantClientTests` which extend `TestWithDb` but only need the database for a couple of tests, so the whole class need not be marked as `@RequiresDB`